### PR TITLE
Draw traffic_signals z15- (vehicle) and z19- (default).

### PR DESCRIPTION
**Pull request by vng**

### CODE REVIEW

**Comment by RedAuburn** [comment]
also see https://github.com/organicmaps/organicmaps/issues/9031 and https://github.com/organicmaps/organicmaps/issues/9030

**Comment by vng** [comment]
Good point. Anyway, this PR makes behavior better than the existing one for sure. But not so good as #9031 :)

**Comment by RedAuburn** [comment]
> Good point. Anyway, this PR makes behavior better than the existing one for sure. But not so good as #9031 :)

agreed, small improvements are still improvements :)

File: data/styles/default/include/priorities_4_overlays.prio.txt [hunk 1]
```diff
@@ -368,6 +368,9 @@ amenity-charging_station-motorcar                   # icon z14- (also has captio
 amenity-fuel                                        # icon z14- (also has caption(optional) z14-)
 === 4270
 
+highway-traffic_signals                             # icon z19-
+=== 4260
```
Comment (pastk) [inline data/styles/default/include/priorities_4_overlays.prio.txt:372]: Too high a priority for the main style? I think it should be very low like 120 which is just under streetside parkings.
WDYT?

File: data/styles/vehicle/include/priorities_4_overlays.prio.txt [hunk 2]
```diff
@@ -174,7 +174,7 @@ amenity-charging_station-motorcar                   # icon z14- (also has captio
 amenity-fuel                                        # icon z12- (also has caption(optional) z12-)
 === 3200
 
-highway-traffic_signals                             # icon z19-
+highway-traffic_signals                             # icon z15-
```
Comment (pastk) [inline data/styles/vehicle/include/priorities_4_overlays.prio.txt:177]: Higher than subway and railway stations even? Hmm..

**Review by pastk — COMMENTED** [review]
What speed (roughly) does z15 correspond to?

Please add some screenshots..

**Comment by vng** [comment]
@pastk I just added new zooms and ran the generator. Feel free to set better priority values.

**Comment by dvdmrtnz** [comment]
Maybe it should only show on the most detailed zoom (z20) in the default map?

How does it the icon look on the default map compared to other icons? Maybe the current icon is too big for the default map? The icon used currently on the vehicle style is 24x24px. Our usual POI icons are 18x18px. See for comparison icons at real size:

  

@organicmaps/design

**Comment by RedAuburn** [comment]
> Maybe the current icon is too big for the default map?

agreed, it was designed to be larger as it was intended to only be shown during routing. we could make a smaller 18px one.

also, it might be good to give them outlines like other traffic icons have:

**Comment by brdbkr** [comment]
> also, it might be good to give them outlines like other traffic icons have:

I think the icon should be as small as possible so it doesn't overlap anything.

**Comment by pastk** [comment]
> agreed, it was designed to be larger as it was intended to only be shown during routing. we could make a smaller 18px one (and keep the larger one for navigation mode)

Also it might make sense to use a bigger icon at high zooms in nav mode and a smaller icon at low zooms like z15-17.

We need screenshots first.

**Comment by dvdmrtnz** [comment]
Redesigned icon for 18x18px size:

**Comment by vng** [comment]
Got regenerated maps. Yes, master icon is very big:
telegram-cloud-photo-size-2-5438617049002797197-y

telegram-cloud-photo-size-2-5438617049002797196-y

**Comment by vng** [comment]
With the new icon looks much better:
telegram-cloud-photo-size-2-5260722849381999941-y

File: data/styles/default/include/priorities_4_overlays.prio.txt (hunk 1 above)
Comment (vng) [reply data/styles/default/include/priorities_4_overlays.prio.txt]: Fixed

File: data/styles/vehicle/include/priorities_4_overlays.prio.txt (hunk 2 above)
Comment (vng) [reply data/styles/vehicle/include/priorities_4_overlays.prio.txt]: Fixed

**Review by pastk — APPROVED** [review]
Not sure if z15 is too early..

Its important to push it through an alpha first to get feedback.

**Comment by vng** [comment]
Merge to run the generator. Zooms will check after the data update.

**Comment by pastk** [comment]
> Can we have the bigger icons while routing maybe?

It was like that in the early versions, check https://github.com/organicmaps/organicmaps/pull/9313#issuecomment-2365411685 (it was too big, too cluttering)

**Pull request merged**